core/src/test/java/fr/sncf/osrd/api/PathfindingTest.java

```diff
@@ -211,8 +212,14 @@ public void incompatibleLoadingGaugeTest() throws Exception {
         );
 
         // Check that we can't go through the infra with a large train
-        assertNull(
-                PathfindingRoutesEndpoint.runPathfinding(infra, waypoints, List.of(TestTrains.FAST_TRAIN_LARGE_GAUGE))
+        assertThrows(
+                NoPathFoundError.class,
+                () -> PathfindingRoutesEndpoint.runPathfinding(
+                        infra,
+                        waypoints,
+                        List.of(TestTrains.FAST_TRAIN_LARGE_GAUGE)
+                ),
+                PathfindingRoutesEndpoint.PATH_FINDING_GAUGE_ERROR
```

Comment:
It seems like this doesn't actually check the error message, which is weird because it should according to the doc. But if you change the string here the test still passes.

Another way to test this is to put the result of `assertThrows` in a variable, then `assertEquals(PathfindingRoutesEndpoint.PATH_FINDING_GAUGE_ERROR, exception.message);`